### PR TITLE
feat(components/google-cloud): Support input type union[str,list[str]]

### PR DESCRIPTION
**Pull request by SinaChavoshi**
Supporting input parameter types of union [ str, list[str] ], for example users can pass in 
```python 
  ds_op = gcc_aip.ImageDatasetCreateOp(
      project=PROJECT_ID,
      display_name='flowers',
      gcs_source='gs://...'
```
or
```python 
  ds_op = gcc_aip.ImageDatasetCreateOp(
      project=PROJECT_ID,
      display_name='flowers',
      gcs_source=['gs://sr1','gs://sr2'...],
```

### CODE REVIEW

**Review by sasha-gitg — APPROVED** [review]
/lgtm

File: components/google-cloud/google_cloud_pipeline_components/aiplatform/remote_runner.py [hunk 1]
```diff
@@ -200,6 +201,22 @@ def prepare_parameters(
                 value = deserializer(value)
             else:
                 value = cast(value, param_type)
+
+            try:
+                # Attempt at converting String to list:
+                # Some parameters accept union[str, sequence[str]]
+                # For these serialization with json is not possible as
+                # component yaml conversion swaps double and single
```
Comment (Ark-kun) [inline components/google-cloud/google_cloud_pipeline_components/aiplatform/remote_runner.py:209]: Can you please add the proof of concept demonstrating the issue?

**Pull request merged**